server/errors/errors.go

```diff
@@ -108,10 +108,14 @@ func InvalidObjectFormat(tuple *openfgapb.TupleKey) error {
 	return status.Error(codes.Code(openfgapb.ErrorCode_invalid_object_format), fmt.Sprintf("Invalid object format for tuple '%s'", tuple.String()))
 }
 
-func UnknownRelation(relation string) error {
+func UnknownRelationWhenWritingAuthzModel(relation string) error {
```

Comment:
Why are we creating two different errors for the same error? I think we should leave this as is but refactor the name to `ErrUnknownRelation`. Any RPC that accepts a `relation` field as an input argument will return this error if the relation they provide is undefined for the authorization model that the request is served against.

Reply:
We also have `RelationNotFound` 😆  They are all the same in meaning but the error message is tailored to each specific context:

`RelationNotFound` -> checks, writes, reads, expands; a tuple is involved
`UnknownRelationWhenWritingAuthzModel` -> writeAuthModel; only relation involved
`UnknownRelationWhenListingObjects` -> listObjects; relation and type involved

Reply:
That's way too overloaded. Let's simplify here. If the relation was not found, regardless of the variation, then lets just return  `ErrRelationNotFound`. Bloating our public API error definitions isn't really buying us anything here, and it's making us define and map more "types" of errors.